controllers/openstackmachine_controller.go

```diff
@@ -430,7 +454,33 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 	return ctrl.Result{}, nil
 }
 
-func (r *OpenStackMachineReconciler) getOrCreate(logger logr.Logger, cluster *clusterv1.Cluster, openStackCluster *infrav1.OpenStackCluster, machine *clusterv1.Machine, openStackMachine *infrav1.OpenStackMachine, computeService *compute.Service, userData string) (*compute.InstanceStatus, error) {
+func (r *OpenStackMachineReconciler) getOrCreatePorts(logger logr.Logger, clusterName string, openStackCluster *infrav1.OpenStackCluster, machinePorts []infrav1.PortOpts, trunkEnabled bool, securityGroups []infrav1.SecurityGroupFilter, instanceTags []string, instanceName string, openStackMachine *infrav1.OpenStackMachine, networkingService *networking.Service) ([]infrav1.PortStatus, error) {
+	// TODO(emilien) implement portsStatus where we check if the machine has ports created and part of the OpenStackMachine.Status
+	// Check `GetInstanceStatusByName` as it's done for instances. For each port found in Status, we might want get the port name and verify there is no duplicate.
```

Comment:
Would it be problematic if there are ports with duplicated name?

Reply:
For CAPO-managed ports, one potential reason why ports could have duplicated names would be that a port is being re-created to replace an old one that wasn't properly removed. I don't see how this can happen with this PR but I might have missed something.
Since we don't have what you proposed in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1788/files#r1440656662 - I guess we're safe now.